mopro-ffi/src/circom/ethereum.rs

```diff
@@ -29,28 +16,29 @@ pub struct ProofCalldata {
 
 // Convert proof to U256-tuples as expected by the Solidity Groth16 Verifier
 // Only supports bn254 for now
-pub fn to_ethereum_proof(proof: Vec<u8>) -> ProofCalldata {
-    let deserialized_proof = deserialize_proof::<Bn254>(proof);
-    let proof = ethereum::Proof::from(deserialized_proof.0);
+pub fn to_ethereum_proof(proof: CircomProof) -> ProofCalldata {
     let a = G1 {
         x: proof.a.x.to_string(),
         y: proof.a.y.to_string(),
+        z: None,
```

Comment:
I remember it would be just `"1".to_string()`?

Reply:
Not actually, `z` represent the point is `infinity` or not. In most of cases, it's `1`, but not always.